bonus.md

```diff
@@ -3,11 +3,17 @@
 Edit this file to answer the following questions. Give examples. You may need to do a bit of research. :)
 
 1. Why is Hash.fetch useful?
+Fetch is useful over just handling with hash[:symbol] because it allows you to handle truthy and falsy defaults that won't work with just using || for defaults
 
 2. What is the disadvantage of single inheritance, and what's a possible solution to that problem?
+A disadvantage of single inheritance is that it can violate encapsulation because you are now raising potential conflicts between methods in the child class and the parent class.
+
 
 3. Why are these classes initialized with an options hash?
+These classes are initialized with an options hash they are initialized with a flexible number of arguments, and we only want to capture certain ones that they have given, or set them to defaults.  It makes our program a lot more flexible.
 
 4. What is the purpose of the private keyword in a class? What does it protect you from and why is that valuable?
+Private means that the method cannot be accessed outside of the class itself, so for example if someone initialized a bank account, and you wanted to store the account number but only return an obscured version, your obfuscation method would be private and the account wouldn't be readable outside of the class, leaving only the obfuscated version available to anyone trying to find that information.  
```

Comment:
Private doesn't actually keep you from being able to access it -- it's more about signaling to people using your code that that method is not intended to be called from outside the class.